ChatNetworkingFirestore/Implementations/Firestore/ChatFirestore.swift

```diff
@@ -316,6 +366,46 @@ extension ChatFirestore {
 
 // MARK: Private methods
 extension ChatFirestore {
+    func getDocumentSnapshot(document: DocumentReference, completion: @escaping (Result<DocumentSnapshot, ChatError>) -> Void) {
+        document.getDocument { [weak self, decoder] (snapshot, error) in
+            self?.networkingQueue.async {
+                if let snapshot = snapshot {
+                    completion(.success(snapshot))
+                } else if let error = error {
+                    completion(.failure(.networking(error: error)))
+                } else {
+                    completion(.failure(.internal(message: "Unknown")))
+                }
+            }
+        }
+    }
+    
+    func getDocuments<T: Decodable>(query: Query, completion: @escaping (Result<[T], ChatError>) -> Void) {
+        query.getDocuments { [weak self, decoder] (documentsSnapshot, error) in
+            self?.networkingQueue.async {
+                if let documentsSnapshot = documentsSnapshot {
+                    var list: [T]
+                    
+                    do {
+                        list = try documentsSnapshot.documents.compactMap {
+                            try $0.decode(to: T.self, with: decoder)
+                        }
+                        
+                    } catch {
+                        logger.log("Couldn't decode document: \(error)", level: .info)
+                        return
```

Comment:
This way you don't get any result even if only one document fails to be decoded. Please get inspired below in `listenToCollection`